### PR TITLE
Adjust app name based on channel, so that beta and stable can both be installed

### PR DESCRIPTION
**Pull request by maxbrunsfeld**
- [x] Change `build` and `install` tasks to name application and command-line launchers based on the channel
  - [x] Mac
  - [x] Windows
  - [x] Linux
- [x] Change package generator tasks to name application and executables based on the channel
  - [x] windows installer
  - [x] debian
  - [x] rpm
- [x] Change atom executable to choose the beta version when invoked as `atom-beta`?
  - [x] Mac & Linux
  - [x] Windows

### CODE REVIEW

**Comment by maxbrunsfeld** [comment]
screen shot 2015-09-22 at 4 29 16 pm

File: atom.sh [hunk 1]
```diff
@@ -74,9 +84,14 @@ if [ $OS == 'Mac' ]; then
 elif [ $OS == 'Linux' ]; then
   SCRIPT=$(readlink -f "$0")
   USR_DIRECTORY=$(readlink -f $(dirname $SCRIPT)/..)
-  ATOM_PATH="$USR_DIRECTORY/share/atom/atom"
-  ATOM_HOME="${ATOM_HOME:-$HOME/.atom}"
 
+  if [ -n "$BETA_VERSION" ]; then
+    ATOM_PATH="$USR_DIRECTORY/share/atom-beta/atom"
+  else
+    ATOM_PATH="$USR_DIRECTORY/share/atom/atom "
```
Comment (50Wliu) [inline atom.sh]: Extra space here

**Comment by maxbrunsfeld** [comment]
rpm

screen shot 2015-09-24 at 10 21 27 am

File: atom.sh (hunk 1 above)
Comment (maxbrunsfeld) [inline atom.sh]: Thanks :eyes:!

**Comment by as-cii** [comment]
Awesome work on this, @maxbrunsfeld!

File: build/Gruntfile.coffee [hunk 2]
```diff
@@ -31,39 +31,52 @@ module.exports = (grunt) ->
   # This allows all subsequent paths to the relative to the root of the repo
   grunt.file.setBase(path.resolve('..'))
 
-  tmpDir = os.tmpdir()
-  appName = if process.platform is 'darwin' then 'Atom.app' else 'Atom'
```
Comment (thomasjo) [inline build/Gruntfile.coffee]: This change is what broke `master` on OS X since this means we end up with an incorrect `shellAppDir`.

**Pull request merged**